### PR TITLE
Release 1.14.0

### PR DESCRIPTION
**Pull request by shirchen**

### CODE REVIEW

**Comment by shirchen** [comment]
@abhinav What do you of releasing structured logging which currently does not have an API to modify the field names (eg some customer may want fields to be emitted as `msg` vs `message`) and then following up with an API in a future release. I think adding structured logging does move the needle forward and will give us some user feedback.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,11 +4,23 @@ All notable changes to this project will be documented in this file.
...
 
-## Unreleased
+## [1.14.0] - 2021-08-12
+### Added
+- Introducing `fx.WithLogger` option that accepts a constructor that returns an 
+  `fxevent.Logger`. This option will override internal logger that will be used
+  to log internal events.
+- Exposing `fxevent` logging interface to implement if using `WithLogger` option
+  when supplying custom logger.
+
```
Comment (sywhang) [inline CHANGELOG.md:14]: We also added a report for hooks runtimes' as part of the error messages displayed on hook timeouts in  https://github.com/uber-go/fx/pull/746.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,21 +6,16 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
-- Fx now emits structured, JSON logs by default. These may be parsed and processed by
-  log ingestion systems.
+- Fx now emits structured JSON logs by default. These may be parsed and
+  processed by log ingestion systems.
 - `fxtest.Lifecycle` now logs to the provided `testing.TB` instead of stderr.
 - `fx.In` and `fx.Out` are now type aliases instead of structs.
-- Standardizing event message tenses and adding new ones: Stopped, Invoking, etc.
-
-### Fixed
-- Fixing a race condition in a unit test.
```
Comment (abhinav) [inline CHANGELOG.md]: (not user facing)

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,21 +6,16 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+  write custom logger implementations.
 
 ### Changed
-- Fx now emits structured, JSON logs by default. These may be parsed and processed by
-  log ingestion systems.
+- Fx now emits structured JSON logs by default. These may be parsed and
+  processed by log ingestion systems.
 - `fxtest.Lifecycle` now logs to the provided `testing.TB` instead of stderr.
 - `fx.In` and `fx.Out` are now type aliases instead of structs.
-- Standardizing event message tenses and adding new ones: Stopped, Invoking, etc.
```
Comment (abhinav) [inline CHANGELOG.md]: (not a change because never released)

**Review by abhinav — APPROVED** [review]
let's do this!

**Review by sywhang — APPROVED** [review]
Yaaay

**Pull request merged**